design/design-doc-observe-only-resources.md

```diff
@@ -0,0 +1,515 @@
+  autoscaler.
+
+Currently, Crossplane does not have a built-in way of observing resources
+without taking ownership of them. There are two workarounds used by the
+community as an interim solution for this gap:
+
+- Using the provider-terraform to observe resources with the help of Terraform
+data sources.
+- Wrapping resources with provider-kubernetes to observe resources managed by
+Crossplane but as a shared object between multiple Compositions.
+
+In this document, we aim to introduce a solution to observe resources with
+Crossplane without taking ownership of them.
+This would allow users to integrate existing cloud resources with the Crossplane
+ecosystem without giving full ownership.
+
+## Goals
+
+- Introduce a way to observe existing resources without taking ownership of them
+in Crossplane.
+- Enable seamless integration of existing cloud resources with the Crossplane
+ecosystem.
+
+### Non-goals
+
+- Partially managing a resource by observing a subset of fields.
+
+This may seem similar to the concept of observing resources, but there is a
+fundamental difference. In this scenario, we want to use certain parameters
+during the creation of the resource, whereas observing resources is intended to
+be a completely read-only operation that should never make any changes to the
+external system, including during the creation process.
+
+## Proposal
+
+To support observing resources without taking ownership, we will introduce a new
+spec named `managementPolicy` to the Managed Resources. We will also deprecate
+the existing `deletionPolicy` in favor of the new spec since they will be
+controlling the same behavior; that is, how should the changes on the CR affect
+the external cloud resource.
+
+This new policy will have the following values:
+
+- `FullControl`(Default): Crossplane will fully manage and control the external
+resource, including deletion when the CR is deleted
+(same as `deletionPolicy: Delete`).
+- `ObserveCreateUpdate`: Crossplane will observe and perform create and update
+operations on the external resource but will not make any deletions
+(same as `deletionPolicy: Orphan`).
+- `ObserveOnly`: Crossplane will only observe the external resource and will not
+make any changes or deletions.
+
+Please note while `ObserveCreateUpdate` may sound verbose, it accurately
+describes the actions that Crossplane will take on the external resource. This
+naming convention also focuses on what Crossplane will do, rather than what it
+won't do, making it more intuitive and extensible for future policy options,
+such as `ObserveDelete`.
+
+As indicated above, `FullControl` and `ObserveCreateUpdate` policies will behave
+precisely the same as the deletion policies we have today, including keeping the
+default behavior the same. We will introduce the new behavior with the
+`ObserveOnly` option, which would be pretty similar to what we have today to
+[import existing managed resources], but instead of starting to manage after
```

Comment:
does this also have the same requirement that import does, where in addition to `external-name` being set, all the required fields of the resource must also be set?

We often hear that is a burden for folks to use import, so will that same burden exist for observeOnly?

Reply:
nevermind, looks like this was addressed later on in the "Schema Changes" section - for k8s 1.25+, users won't have to include the required fields.  Hooray!!